lib/exclusive-lock.js

```diff
@@ -0,0 +1,162 @@
+'use strict'
+
+const {EventEmitter} = require('events')
+const Ajv = require('ajv')
+const schemas = require('./schemas/index.js')
+
+const DEFAULT_LOCK_TTL_MS = 3000
+const DEFAULT_LOCK_REFRESH_MS = 1000
+const REQUIRED_MIN_MS_DIFF = 500
+const DEFAULT_LOCK_CONTENTS = '1'
+const DEFAULT_KEY_PREFIX = 'exclusive-lock'
+const SLUGIFY_REGEX = /[\s\W]+/g
+
+const ajv = new Ajv()
+const validate = ajv.compile(schemas.instantiation_options)
+
+class ExclusiveLock extends EventEmitter {
+  #key
+
+  constructor(opts) {
+    super()
+    opts = opts ?? {}
+
+    let {log} = opts
+    if (!log) {
+      log = require('abstract-logging')
+    }
```

Comment:
You're using the `.child` method elsewhere in the code. It needs to be added to the logger:

```suggestion
    if (!log) {
      log = require('abstract-logging')
      log.child = () => log
    }
```

Reply:
Where?  This main code isn't using a child.  The version in the test action `cache-connection.js`  is an actual pino instance made just for testing.

Reply:
Oh. Carry on.